### DISCUSSION TITLE
Win weight

### DISCUSSION THREAD

**Discussion by MikaelMollberg** [OpenDebates/openskill.py#102] [Q&A] [answered]
Hi, I am looking for a rating system that can be used with tournaments using consolation/placement brackets.

Consolation brackets are for players that have lost once and been eliminated from the main bracket, for example when playing for third place. Since the wins in the lower brackets are "worth less" they need to have lower effect on the ranking. Is this something that is possible with OpenSkill? What currently happens is that mu of 4th place is lower than the mu of 5th place because the 5th place has only lost once, but lost earlier in the tournament while 4th place has lost twice.

**Comment by vivekjoshy** [comment, marked as answer]
Weights are currently not supported due to lack of data. See https://github.com/OpenDebates/openskill.py/issues/29

> What currently happens is that mu of 4th place is lower than the mu of 5th place because the 5th place has only lost once, but lost earlier in the tournament while 4th place has lost twice.

You shouldn't be using, `mu` alone to make your leaderboard. A leaderboard should be constructed using the [`ordinal`](https://openskill.me/en/stable/manual.html#ranks) method. or should be based on a formula that uses both `sigma` and `mu`.

**Reply by MikaelMollberg** [reply]
Thank you for the reply. That's my bad only using mu, I'm still getting my grasps on this and my memories of my statistics course has been lost with time.

What kind of data would you need? I have records of tournaments played with games, sets, points and placement for all teams in the tournament if that helps. Up to 17 teams in a tournament with two players per team.

**Reply by vivekjoshy** [reply]
Any games where player weights make sense. Small sample sizes are usually not good enough as only a small subset of them end up relevant to the benchmarks. For example datasets please take a look [here](https://github.com/OpenDebates/openskill.py/tree/main/benchmark/data). Ideally the data should contain timestamps, ranks, abilities and so on. Something one can use as weights.

**Reply by MikaelMollberg** [reply]
In this case it's not weight on the player, but on the match itself. The importance of the match is lower and should have a smaller effect on the ranking.

If we consider a tournament between players only, the following happens to the 3rd and 4th place ranks.

I have put the ordinal below the player in the image.

![diagram2](https://github.com/OpenDebates/openskill.py/assets/5606160/9cdb721b-645c-464e-a354-35ca6b475cc6)

```
1st Player5 5.045
2nd Player1 3.159
3rd Player8 3.288
4th Player4 -0.479
```

The player that finished 2nd has a value lower than the player that got 3rd place. This is the problem I am looking for a solution for.

**Reply by vivekjoshy** [reply]
I think you're misunderstanding what a rating system is for. It's not meant to create leaderboards for tournaments. It meant to assess and matchup players without having to go through every permutation. In a tournament, you already know wins against who and how to give out ranks. But in online MOBA games or chess, having every player play against each other would take a long time and the players will just keep getting randomly matched with weaker players more often. You can think of the rating a player has as a cutoff point for certain titles (Grandmaster, master etc) so that these titles play each other. There is no advantage for a high ranked player to play a low ranked player since it's a waste of time for the high ranked player. If you want to use tournaments to rank players, a round robin is more effective for a rating system that just throwing random strangers into a tournament. Most tournaments like FIDE will using special pairing algorithms or assignment algorithms like [Swiss-style pairings](https://en.wikipedia.org/wiki/Swiss-system_tournament) or the [Hungarian algorithm](https://en.wikipedia.org/wiki/Hungarian_algorithm).

**Comment by MikaelMollberg** [comment]
I understand. I was hoping to find a system that can work for many tournaments over a season and provide ranks for players, especially if players change teams between tournaments. The same applies here that not all players play against each other which is why I'm trying to find some system that can work for this use case. If you don't see this as an alternative I thank you for your time and I'll look elsewhere.

**Reply by vivekjoshy** [reply]
I am not saying it's not useful for tournaments. I'm saying it forms only part of a tournament. You can still rate players in a tournament. Just remember not to display it as a rank for who won the tournament. It should be used more as a matchmaking tool during a rating period. For instance, here is a typical usage of rating systems where the ratings are used to determine winners:

Season 1: Using Swiss Tournament.  
- $1^{st}$ Round: 
- - All players are unrated, and players are randomly paired with each other such that no pair plays twice.
- - Player ratings updated using OpenSkill.

- $2^{nd}$ Round
- - Player ratings used to decide who to pair up. Low rating players are paired with high rating players to speed up eliminations.
- - Player ratings updated using OpenSkill.
- - Eliminated players don't move to the next round.

- $3^{rd}$ Round

- - Players paired using Monrad, Dutch etc.
- - Player ratings updated using OpenSkill.
- - Eliminated players don't move to the next round.

- $(k - 1)^{th}$ Round (Second Last Round)
- - Players ranked using Monrad, Dutch
- - Player ratings updated using OpenSkill.
- - Winning players move to the next round.

- $k^{th}$ Round (Final Round)
- - Player ratings updated using OpenSkill.
- - Players with highest OpenSkill wins.
